### ISSUE TITLE
Plugin should not operate inside of code blocks or katex blocks

### ISSUE THREAD

**Issue by CalebJohn** [CalebJohn/joplin-rich-markdown#20]
Basically this means that the extra highlighting options should be disabled (e.g. `cm-rm-header-token`).

I think the best way to achieve this will be by updating the main app to leave tokens when in these sections.

**Comment by olbotta** [comment]
yes this plugin should not operate inside katex blocks, the problems are with ^, like this:
![image](https://user-images.githubusercontent.com/29437446/167423193-30f9d4ce-037f-40c6-9c38-43543fc85844.png)
